src/gui/tray/ActivityItemActions.qml

```diff
@@ -46,10 +46,11 @@ RowLayout {
             imageSource: model.modelData.imageSource ? model.modelData.imageSource + root.adjustedHeaderColor : ""
             imageSourceHover: model.modelData.imageSourceHovered ? model.modelData.imageSourceHovered + UserModel.currentUser.headerTextColor : ""
 
-            textColor: imageSource !== "" ? root.adjustedHeaderColor : Style.ncTextColor
-            textColorHovered: imageSource !== "" ? UserModel.currentUser.headerTextColor : Style.ncTextColor
+            textColor: imageSource !== "" || primary ? root.adjustedHeaderColor : Style.ncTextColor
+            textColorHovered: imageSource !== "" || primary ? UserModel.currentUser.headerTextColor : Style.ncTextColor
 
             bold: primary
```

Comment:
IMO this should now be decided internally by the `ActivityActionButton`, where `property bool bold: primaryButton`, as we never set it to anything different here